lib/sqlalchemy/orm/attributes.py

```diff
@@ -270,6 +270,8 @@ class InstrumentedAttribute(QueryableAttribute):
     """
 
     def __set__(self, instance, value):
+        if self.prop.raise_on_set is True:
```

Comment:
tests for this kind of thing start in test/orm/test_attributes.py, then just a quick test for the columnproperty aspect likely in test/orm/test_mapper.py.  these are old test suites so you'd have to step over all the crufty old testing patterns.